src/v/cluster/self_test/diskcheck.cc

```diff
@@ -193,29 +193,29 @@ ss::future<metrics> diskcheck::do_run_benchmark(ss::file& file) {
 template<diskcheck::read_or_write mode>
 ss::future<> diskcheck::run_benchmark_fiber(
   ss::lowres_clock::time_point start, ss::file& file, metrics& m) {
-    auto buf = ss::allocate_aligned_buffer<char>(
-      _opts.request_size, _opts.alignment());
-    random_generators::fill_buffer_randomchars(buf.get(), _opts.request_size);
+    const auto buf_len = std::min(_opts.request_size, 128_KiB);
+    auto buf = ss::allocate_aligned_buffer<char>(buf_len, _opts.alignment());
+    random_generators::fill_buffer_randomchars(buf.get(), buf_len);
+
+    std::vector<iovec> iov;
+    iov.reserve(_opts.request_size / buf_len);
+    for (size_t offset = 0; offset < _opts.request_size; offset += buf_len) {
+        size_t len = std::min(_opts.request_size - offset, buf_len);
+        iov.push_back(iovec{buf.get(), len});
```

Comment:
should we regenerate with new random data here? potentially to avoid making compression more effective?

Reply:
There's no compression option in self_test::diskcheck ,  so i think its cool